File: compatibility-tests/src/test/java/nnl/rocks/kactoos/text/TextBase64Test.java
/**
 * The MIT License (MIT)
 * <p>
 * Copyright (c) 2017-2018 Yegor Bugayenko
 * <p>
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * <p>
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 * <p>
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package nnl.rocks.kactoos.text;

import nnl.rocks.kactoos.matchers.TextHasString;
import org.hamcrest.MatcherAssert;
import org.junit.Test;

/**
 * Test case for  {@link TextBase64}.
 * @author Ilia Rogozhin (ilia.rogozhin@gmail.com)
 * @version $Id: 3090aead405171c002be4df590001bd80aead182 $
 * @since 0.20.2
 */
public final class TextBase64Test {

    /**
     * Check text encodes using the Base64 encoding scheme.
     */
    @Test
    public void checkEncode() {
        MatcherAssert.assertThat(
            "Can't encodes text using the Base64 encoding scheme",
            new TextBase64(
                "Hello!"
            ),
            new TextHasString(
                "SGVsbG8h"
            )
        );
    }
}

File: kactoos-jvm/src/test/kotlin/nnl/rocks/kactoos/text/TextBase64Test.kt


package nnl.rocks.kactoos.text

import nnl.rocks.kactoos.matchers.TextHasString
import org.hamcrest.MatcherAssert
import org.junit.Test

import java.io.IOException

/**
 * Test case for  [nnl.rocks.kactoos.text.TextBase64].
 * @author Ilia Rogozhin (ilia.rogozhin@gmail.com)
 * @version $Id: 3090aead405171c002be4df590001bd80aead182 $
 * @since 0.20.2
 */
class TextBase64Test {

    /**
     * Check text encodes using the Base64 encoding scheme.
     * @throws IOException If fails.
     */
    @Test
    @Throws(IOException::class)
    fun checkEncode() {
        MatcherAssert.assertThat(
                "Can't encodes text using the Base64 encoding scheme",
                TextBase64(
                        "Hello!"
                ),
                TextHasString(
                        "SGVsbG8h"
                )
        )
    }
}
